examples/equilibria/fem_fixed_boundary/fem_magnetostatic_2D_single_coil.py

```diff
@@ -0,0 +1,269 @@
+
+# Import necessary module definitions.
+
+# %%
+
+import os
+
+import dolfin
+import matplotlib.pyplot as plt
+import numpy as np
+
+import bluemira.geometry.tools as tools
+import bluemira.magnetostatics.greens as greens
+from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.file import get_bluemira_root
+from bluemira.equilibria.fem_fixed_boundary.fem_magnetostatic_2D import (
+    FemMagnetostatic2d,
+)
+from bluemira.equilibria.fem_fixed_boundary.utilities import ScalarSubFunc, b_coil_axis
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.placement import BluemiraPlacement
+from bluemira.mesh import meshing
+
+HAS_MSH2XDMF = False
+try:
+    from bluemira.utilities.tools import get_module
+
+    msh2xdmf = get_module(
+        os.path.join(get_bluemira_root(), "..", "msh2xdmf", "msh2xdmf.py")
+    )
+
+    HAS_MSH2XDMF = True
+except ImportError as err:
+    print(f"Unable to import msh2xdmf, dolfin examples will not run: {err}")
+
+# %%[markdown]
+
+# # Creation of the geometry
+
+# Definition of coil and enclosure parameters
+
+# %%
+r_enclo = 100
+lcar_enclo = 0.5
+
+rc = 5
+drc = 0.01
+lcar_coil = 0.01
+
+# %%[markdown]
+
+# create the coil (rectangular cross section) and set the mesh options
+# Note: just to simulate a "real" application, enclosure is generated into the xz
+# plane, but then its placement is changed in order to be in the xy plane (due to a
+# limitation in the importing of the mesh with msh2xdmf for which only the first 2
+# column are considered when importing a 2D mesh)
+
+# %%
+
+poly_coil = tools.make_polygon(
+    [[rc - drc, rc + drc, rc + drc, rc - drc], [0, 0, 0, 0], [-drc, -drc, +drc, +drc]],
+    closed=True,
+    label="poly_enclo",
+)
+poly_coil.change_placement((BluemiraPlacement(axis=[1.0, 0.0, 0.0], angle=-90)))
+
+poly_coil.mesh_options = {"lcar": lcar_coil, "physical_group": "poly_coil"}
+coil = BluemiraFace(poly_coil)
+coil.mesh_options = {"lcar": lcar_coil, "physical_group": "coil"}
+
+# %%[markdown]
+
+# create the enclosure (rectangular cross section) and set the mesh options
+
+# %%
+poly_enclo = tools.make_polygon(
+    [[0, r_enclo, r_enclo, 0], [0, 0, 0, 0], [-r_enclo, -r_enclo, r_enclo, r_enclo]],
+    closed=True,
+    label="poly_enclo",
+)
+poly_enclo.change_placement((BluemiraPlacement(axis=[1.0, 0.0, 0.0], angle=-90)))
+
+poly_enclo.mesh_options = {"lcar": lcar_enclo, "physical_group": "poly_enclo"}
+enclosure = BluemiraFace([poly_enclo, poly_coil])
+enclosure.mesh_options = {"lcar": lcar_enclo, "physical_group": "enclo"}
+
+# %%[markdown]
+
+# create the different components
+
+# %%
+c_universe = Component(name="universe")
+c_enclo = PhysicalComponent(name="enclosure", shape=enclosure, parent=c_universe)
+c_coil = PhysicalComponent(name="coil", shape=coil, parent=c_universe)
+
+# %%[markdown]
+
+# # Mesh
+
+# Create the mesh (by default, mesh is stored in the file Mesh.msh")
+
+# %%
+
+m = meshing.Mesh()
+m(c_universe, dim=2)
```

Comment:
to remove the gmsh log print, one can set the terminal = 0 (instead of 1, default value). I used 1 as the default value cause I had to debug a lot, but know it is not anymore necessary. Modified to 0. Moreover, I have directed the gmsh log info to a logfile (i.e. gmsh.log).